ydb/mvp/oidc_proxy/oidc_protected_page_nebius.cpp

```diff
@@ -105,6 +123,43 @@ void THandlerSessionServiceCheckNebius::ExchangeSessionToken(const TString sessi
     Become(&THandlerSessionServiceCheckNebius::StateExchange);
 }
 
+void THandlerSessionServiceCheckNebius::ExchangeImpersonatedToken(const TString sessionToken, const TString impersonatedToken, const NActors::TActorContext& ctx) {
+    LOG_DEBUG_S(ctx, EService::MVP, "Exchange impersonated token");
+    tokenExchangeType = ETokenExchangeType::ImpersonatedToken;
+    NHttp::THttpOutgoingRequestPtr httpRequest = NHttp::THttpOutgoingRequest::CreateRequestPost(Settings.GetExchangeEndpointURL());
+    httpRequest->Set<&NHttp::THttpRequest::ContentType>("application/x-www-form-urlencoded");
+
+    TMvpTokenator* tokenator = MVPAppData()->Tokenator;
+    TString token = "";
+    if (tokenator) {
+        token = tokenator->GetToken(Settings.SessionServiceTokenName);
+    }
+    httpRequest->Set("Authorization", token); // Bearer included
+    TStringBuilder body;
+    body << "grant_type=urn:ietf:params:oauth:grant-type:token-exchange"
+            << "&requested_token_type=urn:ietf:params:oauth:token-type:access_token"
+            << "&subject_token_type=urn:ietf:params:oauth:token-type:jwt"
+            << "&subject_token=" << impersonatedToken
+            << "&actor_token=" << sessionToken
+            << "&actor_token_type=urn:ietf:params:oauth:token-type:session_token";
+    httpRequest->Set<&NHttp::THttpRequest::Body>(body);
+
+    ctx.Send(HttpProxyId, new NHttp::TEvHttpProxy::TEvHttpOutgoingRequest(httpRequest));
+
+    Become(&THandlerSessionServiceCheckNebius::StateExchange);
+}
```

Comment:
`ExchangeImpersonatedToken` differs from `ExchangeSessionToken` only body building. May be move the general part of these functions to separate function?

Reply:
done